### ISSUE TITLE
State performance measures needed

### ISSUE THREAD

**Issue by schalifoux** [ecsion/eSeaGrant-Client-Bug-Tracker#139]
Programs need the ability to add state performance measures (these were made optional by the National Sea Grant Office in 2014. MIT doesn’t have any which may be why this hasn’t been addressed, but other states, including N.H., do still use them). I’ve attached N.H. Sea Grant’s just as an FYI to show what performance measures are.
Neither the custom form nor the metrics work for what is needed, although the metrics are close. I also think that it is important to have a separate category to distinguish between metrics and performance measures rather than just modifying the metrics section to include the performance measure information and call them all metrics. Metrics and performance measures are distinct in the eyes of the national office and there are other distinctions as well.

Fields that are needed (and need to be displayed to PI) are:
Focus Area (picklist) (configured by admin)
Text of state performance measure (up to 250-300 chars for N.H.’s) (configured by admin)
Target (configured by admin)
Value (entry field for PI)
Comment (entry field for PI)

Like with the metrics, we should be able to select which performance measures to include in a call (i.e., different calls will include different performance measures), and there should be one item (e.g., State Performance Measures) on the left-hand side bar in the call, which, when opened, displays all of the relevant performance measures to the PI (same functionality as metrics).

An example of configuring a performance measure:
Focus Area: Environmental Literacy and Workforce Development
Performance Measure: Number of participants in N.H. Sea Grant formal and informal education programs per year who demonstrate measureable perceived increases in environmental literacy.
Target: 5000
Sequence: 10

An example of what the PI would see:
Environmental Literacy and Workforce Development
Number of participants in N.H. Sea Grant formal and informal education programs per year who demonstrate measureable perceived increases in environmental literacy. 
Target = 5000
Value ________________
Comment ___(expandable text box)____________________________________

[FINAL 2014-17 NHSG State PM and Targets.docx](https://github.com/ecsion/eSeaGrant-Client-Bug-Tracker/files/189852/FINAL.2014-17.NHSG.State.PM.and.Targets.docx)

**Comment by ecsion** [comment]
This is new and will take time. Looking into this.

**Comment by kdez** [comment]
Sue, It looks to me like you should be able to build this in "Custom Forms" no?  It **seems** doable....

However!  You also have the option of having Ecsion build you a "Standard Module" just for your state reporting that is customized just the way you  want it. But it will cost you $$.

**Comment by schalifoux** [comment]
Hi Kathy,

No, I can’t do it in Custom Forms. I had mentioned this limitation to you quite a while ago when looking at the system as it stood then. There are a few issues – target doesn’t display, no focus area, and each measure would have to be a separate form and would display separately with its 200+ characters on the left side menu which would be overwhelming and just doesn’t make sense.

It’s really the same functionality as metrics with a couple extra fields and named “Performance Measures” to distinguish from Metrics.

This isn’t custom state reporting – all states that have performance measures have this need; it’s part of PIER.

Thanks,
Sue

**Comment by kdez** [comment]
Sue can you send me a pdf of your state form?  I want to see what’s involved….

**Comment by kdez** [comment]
Wait a minute – all states?  We are doing our reporting now and have been able to capture everything we need in order to feed PIER.  Do you want to get in and see our set up?

K.

**Comment by schalifoux** [comment]
All states that have state performance measures.

**Comment by schalifoux** [comment]
It’s not a state form. Until the 2014-17 Strategic Plan, NSGO required all states to have state performance measures and objectives that they reported against in PIER in addition to the national performance measures. With the 2014-17 strategic planning cycle, NSGO made it optional for states to have performance measures. MIT chose not to include them. Other states, like N.H., did include them. They are not for the state of N.H. – they align with the national SG Strategic Plan and are for entry into PIER. I’ve attached the table that includes all of our measures. The PI’s need to respond to each measure with a number and a comment.

Thanks,
Sue

From: Chalifoux, Sue
Sent: Monday, March 28, 2016 10:33 AM
To: 'ecsion/eSeaGrant-Client-Bug-Tracker' <reply@reply.github.com>
Cc: Katharine M de Zengotita <kdez@mit.edu>
Subject: RE: [eSeaGrant-Client-Bug-Tracker] State performance measures needed (#139)

Hi Kathy,

No, I can’t do it in Custom Forms. I had mentioned this limitation to you quite a while ago when looking at the system as it stood then. There are a few issues – target doesn’t display, no focus area, and each measure would have to be a separate form and would display separately with its 200+ characters on the left side menu which would be overwhelming and just doesn’t make sense.

It’s really the same functionality as metrics with a couple extra fields and named “Performance Measures” to distinguish from Metrics.

This isn’t custom state reporting – all states that have performance measures have this need; it’s part of PIER.

Thanks,
Sue

**Comment by kdez** [comment]
Okay, I just asked Kathryn Baltes about this, since she knows PIER and since MIT doesn’t do state performance measures. She suggests one Custom Form for each Focus Area.

I would be happy to try to do a clean version of this, and (if it actually is doable) maybe I should demo the build on a joinme call with others – can you send me your state measurables questions/form?

Thanks,
K

**Comment by kdez** [comment]
Thanks, got it – my last crossed your email in the ether – let me work with this targets list a bit….

**Comment by schalifoux** [comment]
I did try that and wasn’t successful with what is needed. The issue with this is that for each measure, the user needs to see both the text and the target (believe me, as someone who has collected reporting for a while now, them having the target is crucial!), then the PI needs to respond with both a comment and a value. Also, we don’t assign all measures for a given focus area to the same PIs – we need to pick and choose measures for each report call. It works very similar to metrics, where the PI’s enter value and note, and we can select which specific metrics go into a report call.

It’s possible I’m missing something, but I could not get the custom form to display the two need value for a measure, even when I tried to use a different form for each separate measure which at least allowed the PI to respond with the two values (number and comment).

Thanks,
Sue
